crates/database/src/solver_competition.rs

```diff
@@ -56,10 +59,17 @@ pub async fn load_by_tx_hash(
     tx_hash: &TransactionHash,
 ) -> Result<Option<LoadCompetition>, sqlx::Error> {
     const QUERY: &str = r#"
-SELECT sc.json, sc.id, s.tx_hash
+WITH competition AS (
+    SELECT sc.id
+    FROM solver_competitions sc
+    JOIN settlements s ON sc.id = s.auction_id
+    WHERE s.tx_hash = $1
+)
+SELECT sc.json, sc.id, COALESCE(ARRAY_AGG(s.tx_hash) FILTER (WHERE s.tx_hash IS NOT NULL), '{}') AS tx_hashes
 FROM solver_competitions sc
 JOIN settlements s ON sc.id = s.auction_id
-WHERE s.tx_hash = $1
+WHERE sc.id = (SELECT id FROM competition)
+GROUP BY sc.id
```

Comment:
Initially, I thought the query could be simplified, but it looks like this is not the case.